python/perf-kernels/streamk/streamk_kernel.py

```diff
@@ -166,25 +146,34 @@ def streamk_gemm(
             B_BASE += BLOCK_SIZE_K * stride_bk
 
         tile_iter = tile_id * iters_per_tile
-        if start_iter == tile_iter:
-            tile_iter_end = tile_iter + iters_per_tile
+
+        if start_iter != tile_iter:
+            rm1 = tl.arange(0, BLOCK_SIZE_M)
+            rn1 = tl.arange(0, BLOCK_SIZE_N)
+            rm1 = tl.max_contiguous(tl.multiple_of(rm1, BLOCK_SIZE_M), BLOCK_SIZE_M)
+            rn1 = tl.max_contiguous(tl.multiple_of(rn1, BLOCK_SIZE_N), BLOCK_SIZE_N)
+            P_ = P + pid * BLOCK_SIZE_M * BLOCK_SIZE_N + rm1[:, None] * BLOCK_SIZE_N + rn1[None, :]
+            tl.store(P_, acc, cache_modifier=".wt")
```

Comment:
Note, for gfx90a the load/stores with `cache_modifiers` do not work. Documented here: https://github.com/ROCm/triton-internal/issues/311